pkgs/by-name/mn/mnamer/package.nix

```diff
@@ -1,31 +1,52 @@
-{ python3Packages, fetchFromGitHub, lib }:
+{
+  python3Packages,
+  fetchFromGitHub,
+  fetchpatch,
+  lib,
+}:
 
 python3Packages.buildPythonApplication rec {
   pname = "mnamer";
-  version = "2.5.4";
+  version = "2.5.5";
+  format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "jkwill87";
     repo = "mnamer";
     rev = version;
-    sha256 = "sha256-fONQq/RboWHFuEFU7HP1ThUpSjOIlkg54c2WlMUKwuk=";
+    sha256 = "sha256-qQu5V1GOsbrR00HOrot6TTAkc3KRasBPDEU7ZojUBio=";
   };
 
+  build-system = with python3Packages; [
+    setuptools
+    setuptools-scm
+  ];
+
+  nativeBuildInputs = with python3Packages; [ pythonRelaxDepsHook ];
+
   propagatedBuildInputs = with python3Packages; [
+    appdirs
     babelfish
+    guessit
     requests
-    appdirs
-    teletype
     requests-cache
-    guessit
+    teletype
   ];
 
-  patches = [
-    # requires specific old versions of dependencies which have been updated in nixpkgs
-    ./remove_requirements.patch
+  pythonRelaxDeps = [
+    "guessit"
+    "requests-cache"
+    "setuptools-scm"
+    "typing-extensions"
+  ];
 
-    # author reads a private property that changed between versions
-    ./update_hack.patch
+  patches = [
+    # https://github.com/jkwill87/mnamer/pull/291
+    (fetchpatch {
+      name = "fix-cached-session-object-error";
+      url = "https://patch-diff.githubusercontent.com/raw/jkwill87/mnamer/pull/291.patch";
```

Comment:
Only merged commits are acceptable to fetch as patch.

Reply:
Oh really? I've never heard this. Is this written somewhere? What's the preferred alternative, checking the patch in to nixpkgs?

Reply:
For small patches we store them in nixpkgs, yes.

I'll see if I can find the guidelines. Really we just need a stable URL, which in practice means we only accept merged commits.

Reply:
Ah I see, thanks. I deleted the other checked in patches and added this as a checked in patch. Hopefully the net removal of patches helps.